### PR TITLE
Create a new top-level module for Galley.

### PR DESCRIPTION
**Pull request by ozevren**
+ Add basic directory structure.
+ Add Makefile support.
+ Add a basic "version" library as a start.

### CODE REVIEW

**Comment by kyessenov** [comment]
why not create it under `istio/pkg`?

**Comment by geeknoid** [comment]
Note that I'll soon be creating istio/pkg/version to be used across the platform so we won't need per-component copies of the same code.

**Comment by xiaolanz** [comment]
@kyessenov Galley contains both an Istio service component and a shared library. So it's not in under pkg/

**Comment by kyessenov** [comment]
Ok, go for it then. I'm thinking that we may need to take it out of normal CI if you guys are going to be churning a lot. A separate run that is not blocking might be better fit.

**Comment by cmluciano** [comment]
@ozevren @xiaolanz Can I be added to the owners file as well?

**Comment by ozevren** [comment]
Of course. Sorry for the oversight. I've updated the PR.

**Comment by ozevren** [comment]
Any chance I can get a /lgtm as well?

**Comment by ozevren** [comment]
@kyessenov it is a good idea to stay out of the normal CI loop for a while (say ~1month), to quickly ramp up things. Thanks.

Any objections? I think initial speed gains in development will be very helpful, especially to deliver something useful for Istio v1.

**Comment by kyessenov** [comment]
I'm fine having split CI, but check with @rshriram and @costinm 
I think it's easy to make galley build non-blocking, but you are still vulnerable to all the flakiness on istio master.

**Comment by rshriram** [comment]
Churn or no churn, let’s please keep master green.

**Comment by cmluciano** [comment]
I think we are hitting an e2e-smoke flake @ozevren

File: galley/pkg/version/version.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Copyright 2018 Istio Authors
```
Comment (mandarjog) [inline galley/pkg/version/version.go]: Remove `galley/pkg/version` in favor of `pkg/version`

File: mixer/pkg/attribute/list.gen.go [hunk 2]
```diff
@@ -3,14 +3,14 @@
 
 package attribute
 
-func GlobalList() ([]string) { 
-    tmp := make([]string, len(globalList))
-    copy(tmp, globalList)
-    return tmp
+func GlobalList() []string {
```
Comment (mandarjog) [inline mixer/pkg/attribute/list.gen.go]: not sure why this file is changing here?

**Comment by mandarjog** [comment]
If you want to just add an empty package, `doc.go` is a fine file to add instead of version.go.

**Comment by geeknoid** [comment]
I've checked in istio/pkg/version so you don't need your own copy.

**Comment by kyessenov** [comment]
It might be almost easier to develop galley in istio/galley if the flakes
continue...
Please don't force merge commits even if it's very tempting to do so.

File: mixer/pkg/attribute/list.gen.go (hunk 2 above)
Comment (ozevren) [reply mixer/pkg/attribute/list.gen.go]: This file keeps sneaking in everytime I do a source format. Removed.

**Comment by ozevren** [comment]
@mandarjog neat. I just removed the version stuff, but make seems to work ok with that. So I'd keep this as is, but if I see any build failures during presubmit I'll just add a doc.go.

**Comment by ozevren** [comment]
Is there a way to merge into Istio main without squashing (thus preserving history)?

Given the journey this simple PR is making through the presubmit system, I am half-tempted to say we should just use another clone repo to get development up and running and use it among Galley folks, followed by a big bang submit.

**Comment by kyessenov** [comment]
That's a question to @geeknoid @rshriram and other technical committee folks :)

**Comment by geeknoid** [comment]
The repo is currently configured for squash merging, but it's easily
flipped for a minute to allow a non-squashing merge.

**Comment by ozevren** [comment]
Can I get another /lgtm?

**Pull request merged**